django/db/migrations/loader.py

```diff
@@ -97,14 +98,16 @@ def load_disk(self):
                 if was_loaded:
                     reload(module)
             self.migrated_apps.add(app_config.label)
-            directory = os.path.dirname(module.__file__)
-            # Scan for .py files
             migration_names = set()
-            for name in os.listdir(directory):
-                if name.endswith(".py"):
-                    import_name = name.rsplit(".", 1)[0]
-                    if import_name[0] not in "_.~":
-                        migration_names.add(import_name)
+            if settings.MIGRATIONS_INCLUDE_PYC:
+                migration_names.update(name for _, name, is_pkg in pkgutil.iter_modules(module.__path__) if not is_pkg)
```

Comment:
Given it's now all `__pycache__`, do we need this workaround? 
(i.e. could we just revert https://github.com/dcwatson/django/commit/51673c146e48ff6230eace39bac18fd555607bee)

Reply:
Perhaps not. I hadn't considered that now that Django is Python 3 only. You can certainly get yourself into the same situation (stale pyc files when switching branches), but now it would certainly be because you went out of your way to compile the legacy pyc files yourself. And if you're doing that you will probably have turned this setting on anyway. So I'm fine with just rolling back the original change.